3_database.py

```diff
@@ -0,0 +1,357 @@
+
+# In[7]:
+
+
+# This is a Python function that connects to a MySQL database using the `mysql.connector` module and executes a given SQL query. 
+
+def execute_query(query): # The function takes a single argument `query`, which is the SQL query to be executed.
+    
+    cnx = mysql.connector.connect(user='root', password='AlumnaAdalab',
+                                host='127.0.0.1', database= 'DB_Ball_API')
+
+    mycursor = cnx.cursor() # It creates a cursor object, executes the query using the cursor, and commits the changes to the database.
+    
+    try: 
+        mycursor.execute(query)
+        cnx.commit()
+
+    except mysql.connector.Error as err:
+        print(err)
+        print("Error Code:", err.errno)
+        print("SQLSTATE", err.sqlstate)
+        print("Message", err.msg)
+    
+    else:
+        mycursor.close()
+        cnx.close() # Finally, the function closes the cursor and the database connection.
+
+
+# `Teams`
+
+# In[13]:
+
+
+teams_table_query = """CREATE TABLE IF NOT EXISTS table_teams (
+id INT NOT NULL,
+abbreviation VARCHAR(255),
+city VARCHAR(255),
+conference VARCHAR(255),
+division VARCHAR(255),
+full_name VARCHAR(255),
+name VARCHAR(255),
+PRIMARY KEY (id));"""
+
+
+# In[14]:
+
+
+execute_query(teams_table_query)
+
+
+# `Games`
+
+# In[15]:
+
+
+games_table_query = """CREATE TABLE IF NOT EXISTS table_games (
+id INT NOT NULL ,
+`date` VARCHAR(255) ,
+home_team_score INT ,
+period INT ,
+postseason VARCHAR(255) ,
+season INT ,
+status VARCHAR(255) ,
+time VARCHAR(255) ,
+visitor_team_score INT ,
+`home_team.id` INT ,
+`home_team.abbreviation` VARCHAR(255) ,
+`home_team.city` VARCHAR(255) ,
+`home_team.conference` VARCHAR(255) ,
+`home_team.division` VARCHAR(255) ,
+`home_team.full_name` VARCHAR(255) ,
+`home_team.name` VARCHAR(255) ,
+`visitor_team.id` INT ,
+`visitor_team.abbreviation` VARCHAR(255) ,
+`visitor_team.city` VARCHAR(255) ,
+`visitor_team.conference` VARCHAR(255) ,
+`visitor_team.division` VARCHAR(255) ,
+`visitor_team.full_name` VARCHAR(255) ,
+`visitor_team.name` VARCHAR(255) ,
+PRIMARY KEY (id));"""
+
+
+# In[16]:
+
+
+execute_query(games_table_query)
+
+
+# `Players`
+
+# In[17]:
+
+
+players_table_query = """CREATE TABLE IF NOT EXISTS table_players (
+id INT NOT NULL ,
+first_name VARCHAR(255),
+height_feet VARCHAR(255),
+height_inches VARCHAR(255),
+last_name VARCHAR(255),
+position VARCHAR(255),
+weight_pounds VARCHAR(255),
+`team.id` INT,
+`team.abbreviation` VARCHAR(255),
+`team.city` VARCHAR(255),
+`team.conference` VARCHAR(255),
+`team.division` VARCHAR(255),
+`team.full_name` VARCHAR(255),
+`team.name` VARCHAR(255),
+PRIMARY KEY (id));"""
+
+
+# In[18]:
+
+
+execute_query(players_table_query)
+
+
+# ## Insert data
+
+# `Teams`
+
+# In[9]:
+
+
+# This code is iterating through each row of a pandas DataFrame using the `iterrows()` method. For each row, it is constructing an SQL query to insert the values of that row into a table.
+for index, row in df_teams.iterrows():
```

Comment:
this is extremely inefficient. For large datasets this is a really bad practice. Try this approach instead
https://datatofish.com/pandas-dataframe-to-sql/
https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.to_sql.html

If you need support let's jump on a call